cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java

```diff
@@ -380,7 +383,8 @@ private void probe(boolean all) {
    */
   private void probe(ImmutableMember member) {
     LOGGER.trace("{} - Probing {}", localMember.id(), member);
-    bootstrapService.getMessagingService().sendAndReceive(member.address(), MEMBERSHIP_PROBE, SERIALIZER.encode(member))
+    bootstrapService.getMessagingService().sendAndReceive(
+        member.address(), MEMBERSHIP_PROBE, SERIALIZER.encode(Pair.of(localMember.copy(), member)), config.getProbeTimeout())
```

Comment:
should we be taking a copy of the member too?

Reply:
The one passed in to the method is copied already. Basically, there's a mutable member and immutable member. The mutable member's `copy()` method just returns an immutable member which is used for replication.